File: pyfastlogging/examples/callback.py
from fastlogging_rs import (
    TRACE,
    DEBUG,
    Logging,
    ConsoleWriterConfig,
    CallbackWriterConfig,
)


def writer_callback(level: int, domain: str, message: str):
    print(f"--> {level} {domain} {message}")


if __name__ == "__main__":
    logger = Logging(
        TRACE,
        "main",
        [
            ConsoleWriterConfig(TRACE, True),
            CallbackWriterConfig(DEBUG, writer_callback),
        ],
    )
    logger.trace("Trace Message")
    logger.debug("Debug Message")
    logger.info("Info Message")
    logger.success("Success Message")
    logger.warning("Warning Message")
    logger.error("Error Message")
    logger.fatal("Fatal Message")
    logger.shutdown()

File: fastlogging/examples/callback.rs
use fastlogging::{CallbackWriterConfig, DEBUG, Logging, LoggingError};

fn writer_callback(level: u8, domain: String, message: String) -> Result<(), LoggingError> {
    println!("CB: {level} {domain}: {message}");
    Ok(())
}

fn main() -> Result<(), LoggingError> {
    let mut logging = Logging::new_unboxed(
        DEBUG,
        "root",
        Some(vec![
            CallbackWriterConfig::new(DEBUG, Some(Box::new(writer_callback))).into(),
        ]),
        None,
        None,
    )
    .unwrap();
    logging.trace("Trace Message".to_string()).unwrap();
    logging.debug("Debug Message".to_string()).unwrap();
    logging.info("Info Message".to_string()).unwrap();
    logging.success("Success Message".to_string()).unwrap();
    logging.warning("Warning Message".to_string()).unwrap();
    logging.error("Error Message".to_string()).unwrap();
    logging.fatal("Fatal Message".to_string()).unwrap();
    logging.shutdown(false).unwrap();
    Ok(())
}
